pkg/providers/azure_test.go

```diff
@@ -0,0 +1,48 @@
+package providers
+
+import (
+	"context"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/tracing"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/subscription/armsubscription"
+	mock_providers "github.com/Azure/draft/pkg/providers/mock"
+	"go.uber.org/mock/gomock"
+	"testing"
+)
+
+func TestGetTenantId(t *testing.T) {
```

Comment:
We should have tests around the different errors/scenarios possible within the getTenantId and listTenant clients